### PR TITLE
EZP-28352: After swapping the location of the content item, the main location of the content item is unassigned	

### PR DESCRIPTION
**Pull request by alongosz**
| Question           | Answer |
| -----------------  | ------ |
| **JIRA issue**     | [EZP-28352](https://jira.ez.no/browse/EZP-28352) |
| **Target version** | `7.x` |
| **Bug fix**        | yes |
| **New feature**    | no |
| **BC breaks**      | no |
| **Tests pass**     | yes |
| **Doc needed**     | no |

This PR replaces invalidating `location-data-<locationId>` with `location-<locationId>` Persistence Cache Tag after Location Swap. 
It turns out that besides Location data, Content is also affected when swapping Main Location.

**TODO**:
- [x] BEFORE MERGE: rebase after #2207 gets merged up.
- [x] Create tests.
- [x] Replace invalidating of `location-data-<locationId>` with `location-<locationId>` tag.
- [x] Update Persistence Cache specification due to discovered issue.
- [x] Confirm Continuous Integration tests are passing.
- [x] Ask for Code Review.

### CODE REVIEW

File: eZ/Publish/API/Repository/Tests/LocationServiceTest.php [hunk 1]
```diff
@@ -1209,6 +1209,43 @@ public function testSwapLocation()
...
+        // load another pre-existing location
+        $secondaryLocation = $locationService->loadLocation($secondaryLocationId);
+
+        // swap the Main Location with a secondary one
+        $locationService->swapLocation($mainLocation, $secondaryLocation);
+
+        // check if Main Location has been updated
+        $mainLocation = $locationService->loadLocation($secondaryLocation->id);
+        self::assertEquals($publishedContent->contentInfo->id, $mainLocation->contentInfo->id);
+        self::assertEquals($mainLocation->id, $mainLocation->contentInfo->mainLocationId);
```
Comment (alongosz) [inline eZ/Publish/API/Repository/Tests/LocationServiceTest.php]: Note: @andrerom This assert and the next one fails if we don't invalidate `location-<locationId>` tag on swap also.

File: eZ/Publish/Core/Persistence/Cache/LocationHandler.php [hunk 2]
```diff
@@ -205,7 +205,14 @@ public function swap($locationId1, $locationId2)
 
         $return = $locationHandler->swap($locationId1, $locationId2);
 
-        $this->cache->invalidateTags(['location-data-' . $locationId1, 'location-data-' . $locationId2]);
+        $this->cache->invalidateTags(
+            [
+                'location-data-' . $locationId1,
+                'location-' . $locationId1,
+                'location-data-' . $locationId2,
+                'location-' . $locationId2,
```
Comment (andrerom) [inline eZ/Publish/Core/Persistence/Cache/LocationHandler.php]: I think if you expire location, then there is no need to expire location-data which is a subset of location.

**Comment by andrerom** [comment]
@alongosz you can now rebase

File: eZ/Publish/API/Repository/Tests/ContentServiceTest.php [hunk 3]
```diff
@@ -5617,27 +5617,6 @@ public function testDeleteTranslationFromDraftThrowsInvalidArgumentException()
         $contentService->deleteTranslationFromDraft($draft->versionInfo, $languageCode);
     }
 
-    /**
-     * Simplify creating custom role with limited set of policies.
-     *
-     * @param $roleName
-     * @param array $policies e.g. [ ['content', 'create'], ['content', 'edit'], ]
-     */
-    private function createRoleWithPolicies($roleName, array $policies)
```
Comment (alongosz) [inline eZ/Publish/API/Repository/Tests/ContentServiceTest.php]: @andrerom merge commit 70a2706 (`6.13` into `7.0`) didn't include this removal done in https://github.com/ezsystems/ezpublish-kernel/commit/be17f3de0ff7a9b6d464df8d1568a446ec1e301a#diff-896f1fe3af332cf2a9e38b343ebe9367L5643 for `6.12`, so adding this here as well. 
I'm not sure why this happened, though :)

File: eZ/Publish/Core/Persistence/Cache/LocationHandler.php (hunk 2 above)
Comment (alongosz) [reply eZ/Publish/Core/Persistence/Cache/LocationHandler.php]: Thanks, I suspected that, but was a bit unsure. Improved in 6d776e3

**Comment by alongosz** [comment]
Rebased, review ping @andrerom @Nattfarinn @webhdx @adamwojs
I think to speed it up, it can be tested by QA already, ping @micszo

File: eZ/Publish/API/Repository/Tests/ContentServiceTest.php (hunk 3 above)
Comment (andrerom) [reply eZ/Publish/API/Repository/Tests/ContentServiceTest.php]: Conflict with removal of TranslationInfo stuff, so lets just remove it here and be done with it.

**Comment by micszo** [comment]
@alongosz checking...

**Comment by micszo** [comment]
@alongosz created the follow-up for investigation https://jira.ez.no/browse/EZP-28663 but I suppose it's for later. 🙂

**Pull request merged**